httomo_backends/methods_database/packages/backends/httomolibgpu/supporting_funcs/recon/algorithm.py

```diff
@@ -180,65 +172,124 @@ def _calc_memory_bytes_LPRec(
         n = n - 1  # dealing with the odd horizontal detector size
         odd_horiz = True
 
+    def debug_print(line_number: int, var_name: str, size_in_bytes: int) -> str:
+        # print(f"{line_number} - {var_name}: {size_in_bytes} B / {size_in_bytes / 1024} KB / {size_in_bytes / 1024 ** 2} MB")
+        pass
+
     eps = 1e-4  # accuracy of usfft
     mu = -np.log(eps) / (2 * n * n)
-    m = int(
-        np.ceil(
-            2
-            * n
-            * 1
-            / np.pi
-            * np.sqrt(
-                -mu * np.log(eps)
-                + (mu * n) * (mu * n) / 4
-            )
-        )
-    )
+    m = int(np.ceil(2 * n * 1 / np.pi * np.sqrt(-mu * np.log(eps) + (mu * n) * (mu * n) / 4)))
 
-    data_c_size = np.prod(0.5 * angles_tot * n) * np.complex64().itemsize
+    center_size = 6144
+    center_size = min(center_size, n * 2 + m * 2)
+    print(f"m: {m}")
+    print(f"center_size: {center_size}")
 
-    fde_size = (
-        0.5 * (2 * m + 2 * n) * (2 * m + 2 * n)
-    ) * np.complex64().itemsize
+    oversampling_level = 2  # at least 2 or larger required
+    ne = oversampling_level * n
+    padding_m = ne // 2 - n // 2
+    print(f"padding_m: {padding_m}")
 
-    c1dfftshift_size = (
-        n * np.int8().itemsize
-    )
-
-    c2dfftshift_slice_size = (
-        np.prod(4 * n * n) * np.int8().itemsize
-    )
+    output_dims = __calc_output_dim_recon(non_slice_dims_shape, **kwargs)
+    if odd_horiz:
+        output_dims = tuple(x + 1 for x in output_dims)
+    print(f"output_dims: {output_dims}")
 
+    in_slice_size = np.prod(non_slice_dims_shape) * dtype.itemsize
+    debug_print(0, "in_slice_size", in_slice_size)
```

Comment:
useful function to have but I reckon we can also comment it out in the estimator itself?